src/headline_grabber/pipeline_steps/filter_max_results.py

```diff
@@ -0,0 +1,25 @@
+from src.headline_grabber.models.pipeline_context import PipelineContext
+from src.headline_grabber.pipeline_steps.pipeline_step import PipelineStep
+from src.headline_grabber.models.headline import Headline
+from typing import List
+
+class FilterMaxResults(PipelineStep):
+    def run(self, context: PipelineContext):
+        if context.user_input.entries is not None:
+            print(context)
+            subject_count = {}
+            fl: List[Headline] = []
+            for headline in context.headlines:
+                
+                if headline.subject.label not in subject_count:
+                    subject_count[headline.subject.label] = 0
+                    
+                if subject_count[headline.subject.label] < context.user_input.entries:
+                    fl.append(headline)
+                    subject_count[headline.subject.label] += 1
+            
```

Comment:
This loop can be written more concisely using python's defaultdict (useful in cases like this where no key has been set yet).

```
from collections import defaultdict

subject_count = defaultdict(int)

for headline in context.headlines:
    if subject_count[headline.subject.label] < context.user_input.entries:
        fl.append(headline)
        subject_count[headline.subject.label] += 1
```

See this thread for more info on defaultdict's usage: https://stackoverflow.com/questions/41455065/understanding-the-use-of-defaultdict-in-python

Reply:
Updated this to use defaultdict.